akka-actor/src/main/scala-2.13-/akka/util/internal/SameThreadExecutionContext.scala

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.util.internal
+
+import akka.actor.ReflectiveDynamicAccess
+import akka.annotation.InternalApi
+
+import scala.concurrent.ExecutionContext
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object SameThreadExecutionContext {
+  def apply(): ExecutionContext = {
+    // we don't want to introduce a dependency on the actor system to use the same thread execution context
+    val dynamicAccess = new ReflectiveDynamicAccess(getClass.getClassLoader)
```

Comment:
a bit unfortunate to create a new ReflectiveDynamicAccess since we have been rather strict about that all reflection should be via `ActorSystem.dynamicAccess`. I understand why that wouldn't be feasible. Would it at least be better to use the classloader of a Scala library class?

Shall we have a catch all fallback and use the previous BatchingExecutor as it was previously defined in `ExecutionContexts.sameThreadExecutionContext`? I'm thinking of Graal native image and such. Maybe a problem for Akka.js also?

Reply:
Hmm, yeah, ok, since this is 2.12 only I think that's fine, let's move our own impl here.

Reply:
For the classloader I'd expect this class to be loaded by the same classloader as the actor system as that is the first place it is used, isn't that what you'd want?

Reply:
Added a fallback with the old inline impl for 2.12 if reflection fails.